src/auth/auth.service.ts

```diff
@@ -1,8 +1,102 @@
-import { Injectable } from '@nestjs/common';
-
+import { ForbiddenException, Injectable, NotImplementedException } from '@nestjs/common';
+import { LoginDto } from './dto/auth';
+import { OrderingCoService } from 'src/ordering-co/ordering-co.service';
+import { ConfigService } from '@nestjs/config';
+import { JwtService } from '@nestjs/jwt';
+import * as argon2 from 'argon2';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { UserService } from 'src/user/user.service';
+import { UserResponse } from 'src/user/dto/user.dto';
+import { Session } from '@prisma/client';
 @Injectable()
 export class AuthService {
-  getHello(): string {
-    return 'Hello World!';
+  constructor(
+    private readonly orderingCoService: OrderingCoService,
+    private configService: ConfigService,
+    private readonly jwt: JwtService,
+    private readonly prisma: PrismaService,
+    private readonly user: UserService,
+  ) {}
+
+  async signIn(loginDto: LoginDto): Promise<UserResponse> {
+    const response = await this.orderingCoService.signIn(loginDto);
+    const tokens = await this.getTokens(response.id, response.email);
+    const user = await this.user.getUserById(response.id);
+
+    if (!user) {
+      const newUser = (await this.user.createUser(response, tokens, loginDto.password)) as UserResponse;
+      return newUser;
+    }
+
+    await this.updateRefreshToken(response.id, tokens.refreshToken, response.accessToken);
+    return UserResponse.createFromUser(user, tokens);
+  }
+
+  signOut() {
+    throw new NotImplementedException();
+  }
+
+  async getTokens(userId: number, email: string): Promise<{ verifyToken: string; refreshToken: string }> {
+    const payload = {
+      sub: userId,
+      email,
+    };
+    const secret = this.configService.get('app_secret');
```

Comment:
as i said that I have change the way the configuration service return, you should use app.secret or not app_secret within get function